### ISSUE TITLE
--no-hw crashes without VA-API drivers

### ISSUE THREAD

**Issue by jbeich** [russelltg/wl-screenrec#15]
VA-API drivers may not be available for some GPUs nor installed by default. Found while testing in a pristine jail with Wayland socket passed through.

```
$ wl-screenrec --no-hw
Using output DP-1
[AVHWDeviceContext @ 0x3f654ad931c0] Failed to initialise VAAPI connection: -1 (unknown libva error).
thread 'main' panicked at 'assertion failed: `(left == right)`
  left: `-5`,
 right: `0`', src/avhw.rs:34:13
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
Abort trap
```

**Comment by russelltg** [comment]
Yeah, that's a good point. The way it's currently written uses vaapi to allocate and do pixel format conversion, even if --no-hw is passed.

Honestly, in this case with no vaapi driver this project has no benefit over wf-recorder, and it would be a decent bit of work to support this. Potential ways this could be done are:
* Allocate the dmabuf surfaces ourselves, and then do our own upload
* Add a path to do shared memory frame copies instead of dmabuf. This is probably the way to go, as it could also enable using this when capturing a small area of a large display (as currently dmabuf copies have to be entire displays)

**Comment by russelltg** [comment]
Note that #90 will "fix" this by allowing you to use vulkan as an allocator instead.

**Comment by russelltg** [comment]
Going to close in favor of #95, which will address this (vulkan is a much more ubiqutious allocator than vaapi)
